src/nemos/regularizer.py

```diff
@@ -336,56 +210,22 @@ def _penalization(
 
         def l2_penalty(coeff: jnp.ndarray, intercept: jnp.ndarray) -> jnp.ndarray:
             return (
-                0.5
-                * self.regularizer_strength
-                * jnp.sum(jnp.power(coeff, 2))
-                / intercept.shape[0]
+                    0.5
+                    * self.regularizer_strength
+                    * jnp.sum(jnp.power(coeff, 2))
+                    / intercept.shape[0]
             )
 
         # tree map the computation and sum over leaves
         return tree_utils.pytree_map_and_reduce(
             lambda x: l2_penalty(x, params[1]), sum, params[0]
         )
 
-    def instantiate_solver(
-        self, loss: Callable, *args: Any, **kwargs: Any
-    ) -> Tuple[SolverInit, SolverUpdate, SolverRun]:
-        """
-        Instantiate the solver with the provided loss function.
-
-        Instantiate the solver with the provided loss function and return callable functions
-        that initialize the solver state, update the model parameters, and run the optimization.
-
-        This method creates a solver instance from jaxopt library, tailored to the specific loss
-        function and regularization approach defined by the Regularizer instance. It also handles
-        the proximal operator if required for the optimization method. The returned functions are
-         directly usable in optimization loops, simplifying the syntax by pre-setting
-        common arguments like regularization strength and other hyperparameters.
-
-        Parameters
-        ----------
-        loss :
-            The original loss function to be optimized.
-
-        Returns
-        -------
-        :
-            A tuple containing three callable functions:
-            - solver_init_state: Function to initialize the solver's state, necessary before starting
-            the optimization.
-            - solver_update: Function to perform a single update step in the optimization process, returning
-            new parameters and state.
-            - solver_run: Function to execute the optimization process, applying multiple updates until a
-            stopping criterion is met.
-        """
-        # this check has be performed here because the penalized loss will
-        # always be a callable independently of which loss is passed!
-        utils.assert_is_callable(loss, "loss")
-
-        def penalized_loss(params, X, y):
+    def penalized_loss(self, loss: Callable) -> Callable:
+        def _penalized_loss(params, X, y):
             return loss(params, X, y) + self._penalization(params)
 
-        return super().instantiate_solver(penalized_loss, *args, **kwargs)
+        return _penalized_loss
 
 
 class ProxGradientRegularizer(Regularizer, abc.ABC):
```

Comment:
We don't need the proximal gradient class anymore, because all regularizer can be used with prox gradient and should have the get_proximal_operator method